src/data/events.ts

```diff
@@ -9,26 +9,66 @@ import EventImage2 from '@assets/events/projects-event-image-2.jpeg';
 import EventImage3 from '@assets/events/projects-event-image-3.jpeg';
 import type { GetImageResult } from 'astro';
 import * as qs from 'qs';
-import type { Event as EventCMS } from './payload-types.ts';
+import type { Address, Event as EventCMS, Media, Team } from './payload-types.ts';
 
-export interface Event {
+export interface EventCalendarEntry {
   title: string;
   date: Date;
   description: string;
-  content_html?: string;
   link?: string;
-  slug?: string;
-  image: {
-    src: Awaited<ReturnType<typeof getImage>>;
-  };
+  image: YVImage;
 }
 
-export async function getNext3Events(): Promise<[Event, Event, Event]> {
-  const image1 = await getImage({ src: thirdEventImage });
-  const image2 = await getImage({ src: calendarCoverImage });
-  const image3 = await getImage({ src: pastEvent });
+export interface EventPage {
+  title: string;
+  start: Date;
+  end: Date;
+  slug: string;
+  content_html: string;
+  heroImage?: YVImage;
+  address: Address;
+  audience: string;
+  cost: string;
+  team: Team;
+  registrationLink: string;
+  timetable: YVImage;
+}
+
+export type YVImage = {
+  src: GetImageResult;
+  alt: string;
+};
 
+export async function getAllEvents(): Promise<EventPage[]> {
+  const response = await fetch(`${process.env.CMS_URL}/api/events`);
```

Comment:
This does not work in the testing environment and is the reason why the `playwright` tests fail.